src/libraries/System.Net.Mail/src/System/Net/Mime/EightBitStream.cs

```diff
@@ -43,74 +46,63 @@ internal EightBitStream(Stream stream, bool shouldEncodeLeadingDots) : this(stre
             _shouldEncodeLeadingDots = shouldEncodeLeadingDots;
         }
 
-        /// <summary>
-        /// Writes the specified content to the underlying stream
-        /// </summary>
-        /// <param name="buffer">Buffer to write</param>
-        /// <param name="offset">Offset within buffer to start writing</param>
-        /// <param name="count">Count of bytes to write</param>
-        /// <param name="callback">Callback to call when write completes</param>
-        /// <param name="state">State to pass to callback</param>
-        public override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback? callback, object? state)
+        public override bool CanRead => false;
+        public override bool CanWrite => BaseStream.CanWrite;
+
+        public override int Read(Span<byte> buffer)
         {
-            ValidateBufferArguments(buffer, offset, count);
+            throw new NotImplementedException();
```

Comment:
In QuotedPrintableStream is `NotSupportedException` being thrown:
```cs
throw new NotSupportedException(SR.ReadNotSupported);
```
But in `EightBitStream` and `QEncodedStream` the NotImplementedException is being thrown.
Shouldn't this be `NotSupportedException` everywhere, since reading is not supported?

Reply:
Good catch, I moved the responsibility for throwing NotImplementedException into DelegatedStream, so now the ReadWrite implementations in derived streams should be unreachable if not supported.